File: 프로그래머스/1/12982. 예산/예산.js
function solution(d, budget) {
    d.sort((a, b) => a - b);

    let total = 0;
    let count = 0;

    for (let i = 0; i < d.length; i++) {
        if (total + d[i] <= budget) {
            total += d[i];
            count++;
        } else {
            break;
        }
    }
    return count;
}

File: 프로그래머스/1/12982. 예산/예산.java
import java.util.*;
class Solution {
    public int solution(int[] d, int budget) {
        int count = 0;
        Arrays.sort(d);
        for(int i=0; i<d.length; i++) {
            if(budget < d[i]) break;
            count++;
            budget -= d[i];
        }
        return count;
    }
}